pysal/esda/moran.py

```diff
@@ -697,6 +698,191 @@ def __quads(self):
+    w : W
+        weight instance assumed to be aligned with y
+    transformation : {'R', 'B', 'D', 'U', 'V'}
+                     weights transformation,  default is row-standardized "r".
+                     Other options include
+                     "B": binary,
+                     "D": doubly-standardized,
+                     "U": untransformed (general weights),
+                     "V": variance-stabilizing.
+    permutations   : int
+                     number of random permutations for calculation of pseudo
+                     p_values
+    geoda_quads    : boolean
+                     (default=False)
+                     If True use GeoDa scheme: HH=1, LL=2, LH=3, HL=4
+                     If False use PySAL Scheme: HH=1, LH=2, LL=3, HL=4
+
+    Attributes
+    ----------
+
+    zx           : array
+                   original x variable standardized by mean and std
+    zy           : array
+                   original y variable standardized by mean and std
+    w            : W
+                   original w object
+    permutations : int
+                   number of random permutations for calculation of pseudo
+                   p_values
+    Is           : float
+                   value of Moran's I
+    q            : array
+                   (if permutations>0)
+                   values indicate quadrat location 1 HH,  2 LH,  3 LL,  4 HL
+    sim          : array
+                   (if permutations>0)
+                   vector of I values for permuted samples
+    p_sim        : array
+                   (if permutations>0)
+                   p-value based on permutations (one-sided)
+                   null: spatial randomness
+                   alternative: the observed Ii is further away or extreme
+                   from the median of simulated values. It is either extremelyi
+                   high or extremely low in the distribution of simulated Is.
+    EI_sim       : float
+                   (if permutations>0)
+                   average value of I from permutations
+    VI_sim       : float
+                   (if permutations>0)
+                   variance of I from permutations
+    seI_sim      : float
+                   (if permutations>0)
+                   standard deviation of I under permutations.
+    z_sim        : float
+                   (if permutations>0)
+                   standardized I based on permutations
+    p_z_sim      : float
+                   (if permutations>0)
+                   p-value based on standard normal approximation from
+                   permutations (one-sided)
+                   for two-sided tests, these values should be multiplied by 2
+
+    Examples
+    --------
+    >>> import pysal as ps
+    >>> import numpy as np
+    >>> np.random.seed(10)
+    >>> w = ps.open(ps.examples.get_path("sids2.gal")).read()
+    >>> f = ps.open(ps.examples.get_path("sids2.dbf"))
+    >>> x = np.array(f.by_col['SIDR79'])
```

Comment:
I'm pretty sure I copied that logic from the docstring in `Moran_Local` :-) `by_col_array()` actually looks more sensible, I didn't know we had that!